src/actions/log.ts

```diff
@@ -0,0 +1,19 @@
+import {PlainReduxAction, ReduxAction} from './redux-action';
+
+export type LogAction = LogErrorsAdd | LogWarningsAdd | LogWarningsRemove | LogErrorsRemove;
+
+export const LOG_ERRORS_ADD = 'LOG_ERRORS_ADD';
+export type LogErrorsAdd = ReduxAction<typeof LOG_ERRORS_ADD, {
+  errors: string[]
+}>;
+
+export const LOG_WARNINGS_ADD = 'LOG_WARNINGS_ADD';
+export type LogWarningsAdd = ReduxAction<typeof LOG_WARNINGS_ADD, {
+  warnings: string[]
+}>;
+
+export const LOG_WARNINGS_REMOVE = 'LOG_WARNINGS_REMOVE';
```

Comment:
`LOG_WARNINGS_ADD` has parameter, but LOG_WARNINGS_REMOVE does not.  Maybe LOG_WARNINGS_CLEAR is a better name?    Same for ERROR.   

I also wonder if we really need two separate CLEAR though?  If they really wanna dismiss warnings and errors, shouldn't we dismiss them in one click.